autopilot/hostfilter.go

```diff
@@ -17,84 +18,118 @@ const (
 	}
 
 	// sanity check - should never happen but this would cause a zero score
 	if h.NetAddress == "" {
-		reasons = append(reasons, "not announced")
+		reasons = append(reasons, errHostNotAnnounced)
 	}
-
 	return len(reasons) == 0, reasons
 }
 
 // isUsableContract returns whether the given contract is usable and whether it
 // can be renewed, along with a list of reasons why it was deemed unusable.
-func isUsableContract(cfg api.AutopilotConfig, s rhp.HostSettings, c api.Contract, bh uint64) (usable bool, refresh bool, renew bool, reasons []string) {
+func isUsableContract(cfg api.AutopilotConfig, s rhpv2.HostSettings, c api.Contract, bh uint64) (usable bool, refresh bool, renew bool, reasons []error) {
+	if isOutOfCollateral(cfg, s, c) {
+		reasons = append(reasons, errContractOutOfCollateral)
+		renew = false
+		refresh = true
+	}
 	if isOutOfFunds(cfg, s, c) {
-		reasons = append(reasons, "out of funds")
+		reasons = append(reasons, errContractOutOfFunds)
 		renew = false
 		refresh = true
 	}
 	if isUpForRenewal(cfg, c.Revision, bh) {
-		reasons = append(reasons, "up for renewal")
+		reasons = append(reasons, errContractUpForRenewal)
 		renew = true
 		refresh = false
 	}
 	if c.Revision.RevisionNumber == math.MaxUint64 {
-		reasons = append(reasons, "max revision number")
+		reasons = append(reasons, errContractMaxRevisionNumber)
 		renew = false
 		refresh = false
 	}
 	if bh > c.EndHeight() {
-		reasons = append(reasons, "expired")
+		reasons = append(reasons, errContractExpired)
 		renew = false
 		refresh = false
 	}
 	usable = len(reasons) == 0
 	return
 }
 
-func isOutOfFunds(cfg api.AutopilotConfig, settings rhp.HostSettings, c api.Contract) bool {
+func isOutOfFunds(cfg api.AutopilotConfig, s rhpv2.HostSettings, c api.Contract) bool {
 	blockBytes := types.NewCurrency64(modules.SectorSize * cfg.Contracts.Period)
-	sectorStoragePrice := settings.StoragePrice.Mul(blockBytes)
-	sectorUploadBandwidthPrice := settings.UploadBandwidthPrice.Mul64(modules.SectorSize)
-	sectorDownloadBandwidthPrice := settings.DownloadBandwidthPrice.Mul64(modules.SectorSize)
+	sectorStoragePrice := s.StoragePrice.Mul(blockBytes)
+	sectorUploadBandwidthPrice := s.UploadBandwidthPrice.Mul64(modules.SectorSize)
+	sectorDownloadBandwidthPrice := s.DownloadBandwidthPrice.Mul64(modules.SectorSize)
 	sectorBandwidthPrice := sectorUploadBandwidthPrice.Add(sectorDownloadBandwidthPrice)
 	sectorPrice := sectorStoragePrice.Add(sectorBandwidthPrice)
 	percentRemaining, _ := big.NewRat(0, 1).SetFrac(c.RenterFunds().Big(), c.TotalCost.Big()).Float64()
 
 	return c.RenterFunds().Cmp(sectorPrice.Mul64(3)) < 0 || percentRemaining < minContractFundUploadThreshold
 }
 
+func isOutOfCollateral(cfg api.AutopilotConfig, s rhpv2.HostSettings, c api.Contract) bool {
+	return isBelowCollateralThreshold(cfg, s, c.RemainingCollateral(s))
+}
+
+func isBelowCollateralThreshold(cfg api.AutopilotConfig, s rhpv2.HostSettings, c types.Currency) bool {
```

Comment:
could use a docstring that explains what we consider too little and that it is related to the initial collateral